doc/architecture/decisions/0004-secret-generation.md

```diff
@@ -0,0 +1,32 @@
+# 4. Secret Generation
+
+Date: 2022-12-07
+
+## Status
+
+Proposed
+
+## Context
+
+An integration recently failed when a new secret was generated containing special characters which was not properly escaped. This highlighted the need to have a documented and standard process for generating new secrets.
+
+## Decision
+
+The following process is to be adopted for secret generation - 
+
+1. Generate a new secret using 1Password. All PlanX developers have access to this tool as standard.
+2. Secrets can contain special characters, as long as they are properly escaped in the `.env` file.
+   - Please see [`dotenv` docs](https://github.com/motdotla/dotenv/blob/master/README.md#what-rules-does-the-parsing-engine-follow) for implementation details
```

Comment:
I think this may be overkill, but we probably do need to think about how we handle secrets and pass them around the application. A validation step could be included in this.

Currently, we call `process.env.YOUR_SECRET` everywhere, which either requires use of the non-null operator (!) everywhere (which is rightly against our linting rules), or multiple `assert()` calls. This can get particularly painful if a secret is used multiple times across files or conditionally.

Certainly something to discuss, but a process that checked the env file once and then re-exposed them could handle this. Something as simple as this would do the job - https://github.com/facebook/flow/issues/1192#issuecomment-272629411